### DISCUSSION TITLE
Best way to create multiple custom image files

### DISCUSSION THREAD

**Discussion by mbilloo** [OE4T/meta-tegra#636] [Q&A] [answered]
Hi,

I'm working on a project for a Jetson Nano that requires a custom partition table. I achieved this by creating my own image type, inheriting from image_types_tegra, overriding the "tegraflash_create_config_tegra210" function, and add my customizations to the flash.xml file. This seems to work OK. The next step is to create image files that will populate some of the custom partitions. I achieved this by overriding the "tegraflash_custom_post" command and simply creating an appropriately sized image filled with zeros (i.e. dd if=/dev/zero ...) and turning it into an ext4 filesystem (i.e. mkfs.ext4). But, this is not sustainable, since won't be able to directly add packages to these images later using this scheme/technique.

Is there a better way to create empty image files besides the "main" rootfs that will make it easier to add packages later?

**Comment by dwalkes** [comment, marked as answer]
Hi @mbilloo 
This probably relates to the discussion in https://github.com/OE4T/meta-tegra/discussions/620 if you haven't found that one yet.

Also regarding creating your own custom image files you might refer to the example from [mender dataimg](https://github.com/mendersoftware/meta-mender/blob/86ecd224c869e9a58228503d02a0de065869fbfd/meta-mender-core/classes/mender-dataimg.bbclass) used in [tegra-demo-distro tegrademo-mender](https://github.com/OE4T/tegra-demo-distro/search?q=dataimg) and [meta-mender-community meta-mender-tegra layer](https://github.com/mendersoftware/meta-mender-community/search?q=dataimg)
